SETUP/check_require_login.php

```diff
@@ -0,0 +1,186 @@
+    "quiz/tuts/tut_p_aeoe_1.php",
+    "quiz/tuts/tut_p_greek_5.php",
+    "quiz/tuts/tut_p_greek_1.php",
+    "quiz/tuts/tut_p_mod2_4.php",
+    "quiz/tuts/tut_p_mod1_1.php",
+    "quiz/tuts/tut_p_basic_2.php",
+    "quiz/tuts/tut_p_old_1.php",
+    "quiz/tuts/tut_p_greek_4.php",
+    "quiz/tuts/tut_p_mod2_5.php",
+    "quiz/tuts/tut_p_fraktur.php",
+    "quiz/tuts/tut_p_greek_3.php",
+    # Stats graphs that we make globally available
+    "stats/default.php",
+    "stats/jpgraph_files/percent_users_who_proof.php",
+    "stats/jpgraph_files/pages_daily.php",
+    "stats/jpgraph_files/cumulative_total_proj_summary_graph.php",
+    "stats/jpgraph_files/users_by_pages_proofed_graph.php",
+    "stats/jpgraph_files/users_by_country.php",
+    "stats/jpgraph_files/round_backlog_trend.php",
+    "stats/jpgraph_files/round_backlog_days.php",
+    "stats/jpgraph_files/round_backlog.php",
+    "stats/jpgraph_files/cumulative_total_proj_graph.php",
+    "stats/jpgraph_files/users_logging_on.php",
+    "stats/jpgraph_files/users_by_language.php",
+    "stats/jpgraph_files/total_proj_graph.php",
+    "stats/jpgraph_files/average_hour_users_logging_on.php",
+    "stats/jpgraph_files/total_pages_by_month_graph.php",
+    "stats/jpgraph_files/new_users.php",
+    "stats/jpgraph_files/cumulative_month_proj.php",
+    "stats/jpgraph_files/tallyboard_deltas.php",
+    "stats/jpgraph_files/curr_month_proj.php",
+    "stats/jpgraph_files/equilibria.php",
+    # Design documentation
+    "styles/style_demo.php",
+    "styles/design_philosophy.php",
+    # Tools
+    "tools/setlangcookie.php",  # allows unauth users to set their language
+    "tools/post_proofers/smooth_reading.php",
+    "tools/proofers/ctrl_frame.php",  # required for unauth quiz access
+];
+
+$files = get_all_php_files("../");
+foreach($files as $file)
+{
+    # If it requires authentication, skip it
+    if(file_requires_auth("../$file"))
+        continue;
+
+    # If it's in the SETUP directory, skip it
+    if(startswith($file, "SETUP/"))
+        continue;
+
+    # If it's on the OK list, skip it
+    if(in_array($file, $ok_files))
+        continue;
+
+    echo "$file does not require authentication and is not on the known-good list\n";
+    exit(1);
+}
+
+function get_all_php_files($basedir)
+{
+    $php_files = [];
+
+    $dir_iter = new RecursiveDirectoryIterator($basedir);
+    $files = new RecursiveIteratorIterator($dir_iter);
+    foreach($files as $file_info)
+    {
+        $file = $file_info->getPathname();
+        if(!endswith($file, ".php"))
+            continue;
+        $php_files[] = str_replace($basedir, "", $file);
+    }
+    return $php_files;
+}
+
+function file_requires_auth($filename)
+{
+    $contents = file_get_contents($filename);
+
+    # "regular" pages use require_login();
+    $login = preg_match("/^require_login\(\);$/m", $contents);
+
+    # crontab pages use requester_is_localhost()
+    $localhost = preg_match("/requester_is_localhost\(\)/m", $contents);
```

Comment:
Side thought: we should maybe define (say) `require_localhost_request()` as a parallel to `require_login()`.

Reply:
I like this idea a lot. Standby for update.

Reply:
And added.